test/http/message.cpp

```diff
@@ -147,9 +146,154 @@ class message_test : public beast::unit_test::suite
         BEAST_EXPECT(m2.headers.exists("h"));
     }
 
+    struct MoveHeaders
+    {
+        bool moved_to = false;
+        bool moved_from = false;
+
+        MoveHeaders() = default;
+
+        MoveHeaders(MoveHeaders&& other)
+            : moved_to(true)
+        {
+            other.moved_from = true;
+        }
+
+        MoveHeaders& operator=(MoveHeaders&& other)
+        {
+            return *this;
+        }
+    };
+
+    void testHeaders()
+    {
+        {
+            using req_type = request_headers<headers>;
+            static_assert(std::is_copy_constructible<req_type>::value, "");
+            static_assert(std::is_move_constructible<req_type>::value, "");
+            static_assert(std::is_copy_assignable<req_type>::value, "");
+            static_assert(std::is_move_assignable<req_type>::value, "");
+
+            using res_type = response_headers<headers>;
+            static_assert(std::is_copy_constructible<res_type>::value, "");
+            static_assert(std::is_move_constructible<res_type>::value, "");
+            static_assert(std::is_copy_assignable<res_type>::value, "");
+            static_assert(std::is_move_assignable<res_type>::value, "");
+        }
+
+        {
+            MoveHeaders h;
+            request_headers<MoveHeaders> r{std::move(h)};
+            BEAST_EXPECT(h.moved_from);
+            BEAST_EXPECT(r.headers.moved_to);
+            request<string_body, MoveHeaders> m{std::move(r)};
+            BEAST_EXPECT(r.headers.moved_from);
+            BEAST_EXPECT(m.headers.moved_to);
+        }
+
+        {
+            MoveHeaders h;
```

Comment:
This scope looks identical to the one above starting in L185.  Am I missing something?